python/cuda_parallel/cuda/parallel/experimental/_cccl.py

```diff
@@ -130,8 +131,10 @@ def _numpy_type_to_info(numpy_type: np.dtype) -> TypeInfo:
     return _numba_type_to_info(numba_type)
 
 
-def _device_array_to_cccl_iter(array) -> Iterator:
-    info = _numpy_type_to_info(array.dtype)
+def _device_array_to_cccl_iter(array: DeviceArrayLike) -> Iterator:
+    if get_stride(array) is not None:
```

Comment:
Only checking whether `strides` is None is not robust. An array library could choose to fully specify the strides even if the array is C-contiguous. Also, this would reject F-contiguous arrays (do we not support it?)

Reply:
Thanks - do you have suggestions for the most robust way to check for contiguity? 

> Also, this would reject F-contiguous arrays (do we not support it?)

I suppose there's no reason not to support them.

Reply:
I took a stab at implementing `is_contiguous` in a more robust way. I tried to take into consideration all the information in the ndarray [docs](https://numpy.org/doc/2.1/reference/arrays.ndarray.html#internal-memory-layout-of-an-ndarray) about strides and contiguity. Your feedback here would be extremely valuable!

https://github.com/NVIDIA/cccl/pull/3233/commits/79b3a2ac6aa2e524c40ed43d05ab92bb49bafd6f